tests/abe/cover_crypt.test.ts

```diff
@@ -1,24 +1,125 @@
+)
+
+// User decryption key with access policy: "Security Level::Medium Secret && Department::MKG"
+const MEDIUM_SECRET_MKG_USER_ACCESS_POLICY =
+  "Security Level::Medium Secret && Department::MKG"
+
+// Plaintext example is: My secret message
+const PLAINTEXT = hexDecode("4d7920736563726574206d657373616765")
+
+// Hybrid encrypted data: ABE attributes are ['Security Level::Low Secret', 'Department::HR', 'Department::FIN']
+const ENCRYPTED_DATA = hexDecode(
+  "e07ffb36e8935ee4c2c7a27402042c332868c647914f908abaf745a78b179a6a0cfc823dc18aefb894e49cf80f3d8868434b822498b17482a3285691d62f6f4101bd54050526d82b8048ed019e73ef818a2d1c61d1bd21b849fbf2721802b4f3d56dfe18f0107a1c0175864568d0cc9aaf8f123e7b4ef7d53196de5649c605d57e1c2db9125da53983fd5f99e997f20a29f4d2cf99d0f4367a3e460416ebb7f282358a762ddf8f0136e8114288004e69488f4496a184cad01555a0183ed4dd880d28317f0d3ec6d93e2f13"
+)
+
+test("Non regression tests", async () => {
+  const { CoverCryptHybridDecryption } = await CoverCrypt();
+
+  const hybridDecryption = new CoverCryptHybridDecryption(TOP_SECRET_MKG_FIN_USER)
+  const cleartext = hybridDecryption.decrypt(ENCRYPTED_DATA)
+
+  expect(cleartext).toEqual(PLAINTEXT);
+})
 
 test("cover_crypt", async () => {
-  const keyGeneration = new CoverCryptKeyGeneration()
-  const demoKeys = new CoverCryptDemoKeys()
-  const hybridEncryption = new CoverCryptHybridEncryption(
-    demoKeys.policy,
-    demoKeys.publicKey
+  const { CoverCryptKeyGeneration } = await CoverCrypt();
+  const keyGenerator = new CoverCryptKeyGeneration();
+
+  const policy = new Policy(
+    [
+      new PolicyAxis(
+        "Security Level",
+        [
+          "Protected",
+          "Low Secret",
+          "Medium Secret",
+          "High Secret",
+          "Top Secret",
+        ],
+        true
+      ),
+      new PolicyAxis("Department", ["R&D", "HR", "MKG", "FIN"], false),
+    ],
+    100
+  )
+
+  await runTests(policy);
+
+  const newPolicy = keyGenerator.rotateAttributes(
+    ["Security Level::Low Secret", "Department::MKG"],
+    policy,
+  )
+
+  await runTests(newPolicy);
+});
+
+/**
+ * Run some encryption/decryption tests
+ * 
+ * @param {Policy} policy policy to use
+ */
+async function runTests(policy: Policy): Promise<void> {
+  const { CoverCryptHybridDecryption, CoverCryptHybridEncryption, CoverCryptKeyGeneration } = await CoverCrypt();
+
+  const keyGenerator = new CoverCryptKeyGeneration();
+  const newMasterKeys = keyGenerator.generateMasterKeys(policy)
+
+  const topSecretMkgFinUser =
+    keyGenerator.generateUserDecryptionKey(
```

Comment:
You can rename to be consistent with the CoverCrypt naming
```suggestion
    keyGenerator.generateUserSecretKey(
```